src/Compilers/CSharp/Portable/Emitter/EditAndContinue/CSharpSymbolMatcher.cs

```diff
@@ -740,6 +740,16 @@ bool isCorrespondingType(TypeWithAnnotations type, TypeWithAnnotations expectedT
             private Symbol? VisitNamedTypeMember<T>(T member, Func<T, T, bool> predicate)
                 where T : Symbol
             {
+                if (member.ContainingType is null)
+                {
+                    // ContainingType is null for synthesized PrivateImplementationDetails helpers.
+                    // For simplicity, these helpers are not reused across generations.
+                    // Instead new ones are regenerated as needed.
+                    Debug.Assert(member is ISynthesizedGlobalMethodSymbol);
+
+                    return null;
```

Comment:
It looks like the PR description says that we are emitting a new copy of a private implementation details class. So, we are not going to end up with duplicate signatures in the same type because of this logic. Correct?